File: trip-service-java/src/main/java/com/cawadev/softwarecraft/katas/tripservice/trip/TripService.java
package com.cawadev.softwarecraft.katas.tripservice.trip;

import com.cawadev.softwarecraft.katas.tripservice.exception.UserNotLoggedInException;
import com.cawadev.softwarecraft.katas.tripservice.user.User;
import com.cawadev.softwarecraft.katas.tripservice.user.UserSession;

import java.util.ArrayList;
import java.util.List;

public class TripService {

    public List<Trip> getTripsByUser(User user) throws UserNotLoggedInException {
        if (getLoggedUser() == null) {
            throw new UserNotLoggedInException();
        }

        return user.isFriendsWith(getLoggedUser())
                ? tripsByUser(user) :
                new ArrayList<>();
    }

    protected List<Trip> tripsByUser(User user) {
        return TripDAO.findTripsByUser(user);
    }

    protected User getLoggedUser() {
        return UserSession.getInstance().getLoggedUser();
    }
}

File: trip-service-kotlin/src/main/kotlin/com/cawadev/softwarecraft/katas/tripservice/trip/TripService.kt
package com.cawadev.softwarecraft.katas.tripservice.trip

import com.cawadev.softwarecraft.katas.tripservice.exception.UserNotLoggedInException
import com.cawadev.softwarecraft.katas.tripservice.user.User

class TripService(private val tripDAO: TripDAO) {

    fun getTripsByUser(
        user: User,
        loggedInUser: User?
    ): List<Trip> {
        if (loggedInUser === null) {
            throw UserNotLoggedInException()
        }

        return if (user.isFriendsWith(loggedInUser)) {
            tripsByUser(user)
        } else emptyList()
    }

    private fun tripsByUser(user: User) = tripDAO.tripsByUser(user)
}
